core/capabilities/triggers/logevent/service.go

```diff
@@ -83,7 +85,8 @@ func (s *TriggerService) Info(ctx context.Context) (capabilities.CapabilityInfo,
 
 // Register a new trigger
 // Can register triggers before the service is actively scheduling
-func (s *TriggerService) RegisterTrigger(ctx context.Context, req capabilities.TriggerRegistrationRequest) (<-chan capabilities.TriggerResponse, error) {
+func (s *TriggerService) RegisterTrigger(ctx context.Context,
+	req capabilities.TriggerRegistrationRequest) (<-chan capabilities.TriggerResponse, error) {
```

Comment:
Is performance a concern for `RegisterTrigger`? If yes, we could probably scope out avoiding channel allocation inside this fn and instead let the caller pass that in the arg? But I do not have the complete context here so I might be missing the bigger picture 👀

Reply:
Hmm.. how would that improve performance?
Besides, this type signature is required by the Capability interface, so this implementation has to adhere to the interface requirements of TriggerCapability